pyro/distributions/empirical.py

```diff
@@ -165,11 +182,23 @@ def variance(self):
         deviation_squared = torch.pow(self._samples - self.mean, 2)
         return self._weighted_mean(deviation_squared)
 
-    def get_samples_and_weights(self):
-        self._finalize()
-        return self._samples, self._log_weights
-
+    @_finalized
+    def get_data(self, flatten=False):
```

Comment:
This name might confuse users for what `data` means in the following context:
```
marginal = inference.run(data).marginal()
marginal.empirical()["a"].get_data()
```
How about `get_support`? or `get_underlined_samples`?

Reply:
I just realized that we have ``.enumerate_support()`` method, so let us use that directly and remove this altogether.